### PR TITLE
Initial implementation of the system UI overlay methods

### PR DESCRIPTION
**Pull request by swift-kim**
Contributes to https://github.sec.samsung.net/f-project/home/issues/47 (internal repo).

Partially implements
- [SystemChrome.setEnabledSystemUIOverlays](https://api.flutter.dev/flutter/services/SystemChrome/setEnabledSystemUIOverlays.html) (deprecated)
- [SystemChrome.setEnabledSystemUIMode](https://api.flutter.dev/flutter/services/SystemChrome/setEnabledSystemUIMode.html) (only `SystemUiMode.manual` is allowed)
- [SystemChrome.restoreSystemUIOverlays](https://api.flutter.dev/flutter/services/SystemChrome/restoreSystemUIOverlays.html)

This change only implements the navigation bar (softkey) part of the system overlays (supports only `SystemUiOverlay.bottom`). The status bar (indicator) part will be implemented in a later commit.

TZSH-Softkey API information: https://docs.tizen.org/application/native/guides/ui/tizen-ws-shell/tzsh-softkey (since Tizen 5.5)

Additional notes:
- Only works for the common profile, because it's the only profile that has a software navigation bar.
- You need to add the following to your `tizen-manifest.xml` and use a platform level distributor certificate to sign the app.<p>
  ```xml
  <feature name="http://tizen.org/feature/ui_service.softkey"/>
  <privileges>
      <privilege>http://tizen.org/privilege/windowsystem.admin</privilege>
  </privileges>
  ```
- Use [WidgetsBindingObserver.didChangeAppLifecycleState](https://api.flutter.dev/flutter/widgets/WidgetsBindingObserver/didChangeAppLifecycleState.html) if you want to restore system UI overlays on app resume.

The `AZURE_BUILD_ID` for this PR is `674`.

### CODE REVIEW

File: shell/platform/tizen/channels/platform_channel_tizen.cc [hunk 1]
```diff
@@ -84,6 +86,87 @@ class FeedbackManager {
...
+  void ShowSoftkey() {
+    if (!tizen_shell_softkey_) {
+      return;
+    }
+    int ret = tzsh_softkey_global_show(tizen_shell_softkey_);
+    if (ret != TZSH_ERROR_NONE) {
+      FT_LOG(Error) << "tzsh_softkey_global_show() failed with error: "
+                    << get_error_message(ret);
+    }
+    is_softkey_shown_ = true;
```
Comment (HakkyuKim) [inline shell/platform/tizen/channels/platform_channel_tizen.cc:129]: `is_softkey_shown` changes after failure, maybe missing return?

File: shell/platform/tizen/channels/platform_channel_tizen.cc [hunk 2]
```diff
@@ -84,6 +86,87 @@ class FeedbackManager {
...
+      return;
+    }
+    // Always show the softkey before hiding it again, to avoid subtle bugs.
+    tzsh_softkey_global_show(tizen_shell_softkey_);
+    int ret = tzsh_softkey_global_hide(tizen_shell_softkey_);
+    if (ret != TZSH_ERROR_NONE) {
+      FT_LOG(Error) << "tzsh_softkey_global_hide() failed with error: "
+                    << get_error_message(ret);
+    }
+    is_softkey_shown_ = false;
```
Comment (HakkyuKim) [inline shell/platform/tizen/channels/platform_channel_tizen.cc:143]: Ditto.

Comment (bwikbs) [reply shell/platform/tizen/channels/platform_channel_tizen.cc:143]: > Ditto.

A word that reminds me someone.. (Never mind!, it's a joke for @bbrto21 :smile: )

File: shell/platform/tizen/channels/platform_channel_tizen.cc (hunk 1 above)
Comment (swift-kim) [reply shell/platform/tizen/channels/platform_channel_tizen.cc]: Sounds reasonable. Thanks!

**Review by HakkyuKim — APPROVED** [review]
LGTM!

**Review by bwikbs — APPROVED** [review]
:+1:

**Pull request merged**